packages/venia-ui/lib/components/CartPage/PriceAdjustments/CouponCode/couponCode.js

```diff
@@ -44,6 +49,11 @@ const REMOVE_COUPON_MUTATION = gql`
             cart {
                 id
                 ...CartPageFragment
+                # If this mutation causes "free" to become available we need to know.
```

Comment:
We _must_ do this for mutations that could cause the cart total to become $0 or change from $0. I'm not sure how to enforce this but I applied it to gift card and coupon mutations. We probably need to add this to the add/remove item mutations as well.

I'd really appreciate thoughts here.